OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/NewCourseOutlineAdapter.java

```diff
@@ -0,0 +1,679 @@
+        return convertView;
+    }
+
+    /**
+     * Adds last accessed course item view in the ListView.
+     *
+     * @param lastAccessedComponent The last accessed component.
+     * @param onClickListener       The listener to invoke when the `view` button is pressed.
+     */
+    public void addLastAccessedView(CourseComponent lastAccessedComponent, View.OnClickListener onClickListener) {
+        // Update the last accessed item, if its already there in the list
+        if (mData.size() >= 2 && mData.get(1).type == SectionRow.LAST_ACCESSED_ITEM) {
+            mData.set(1, new SectionRow(SectionRow.LAST_ACCESSED_ITEM, lastAccessedComponent, onClickListener));
+        } else {
+            // Add it otherwise
+            mData.add(1, new SectionRow(SectionRow.LAST_ACCESSED_ITEM, lastAccessedComponent, onClickListener));
+        }
+        notifyDataSetChanged();
+    }
+
+    public ViewHolder getTag(View convertView) {
+        ViewHolder holder = new ViewHolder();
+        holder.rowType = (IconImageView) convertView
+                .findViewById(R.id.row_type);
+        holder.rowTitle = (TextView) convertView
+                .findViewById(R.id.row_title);
+        holder.rowSubtitle = (TextView) convertView
+                .findViewById(R.id.row_subtitle);
+        holder.rowSubtitleDueDate = (TextView) convertView
+                .findViewById(R.id.row_subtitle_due_date);
+        holder.rowSubtitleIcon = (IconImageView) convertView
+                .findViewById(R.id.row_subtitle_icon);
+        holder.rowSubtitleIcon.setIconColorResource(R.color.edx_brand_primary_base);
+        holder.noOfVideos = (TextView) convertView
+                .findViewById(R.id.no_of_videos);
+        holder.bulkDownload = (IconImageView) convertView
+                .findViewById(R.id.bulk_download);
+        holder.bulkDownload.setIconColorResource(R.color.edx_brand_gray_accent);
+        holder.numOfVideoAndDownloadArea = (LinearLayout) convertView
+                .findViewById(R.id.bulk_download_layout);
+        holder.rowSubtitlePanel = convertView.findViewById(R.id.row_subtitle_panel);
+        holder.halfSeparator = convertView.findViewById(R.id.row_half_separator);
+        holder.wholeSeparator = convertView.findViewById(R.id.row_whole_separator);
+
+        // Accessibility
+        ViewCompat.setImportantForAccessibility(holder.rowSubtitle, ViewCompat.IMPORTANT_FOR_ACCESSIBILITY_NO);
+
+        return holder;
+    }
+
+    public static class ViewHolder {
+        IconImageView rowType;
+        TextView rowTitle;
+        TextView rowSubtitle;
+        TextView rowSubtitleDueDate;
+        IconImageView rowSubtitleIcon;
+        IconImageView bulkDownload;
+        TextView noOfVideos;
+        LinearLayout numOfVideoAndDownloadArea;
+        View rowSubtitlePanel;
+        View halfSeparator;
+        View wholeSeparator;
+    }
+
+    public static class SectionRow {
+        public static final int ITEM = 0;
```

Comment:
Adapter's `getView` and `getItemViewType` don't play well along with enums, so current implementation is correct. We don't need enum here since we have gotten rid of the `getViewTypeCount` function as well (which was unused).

Reply:
any reference ?

Reply:
Umm, it is possible but for time being, refactoring needs proper testing afterwards and will go out of scope. So skipping it!